src/model_filereading.cpp

```diff
@@ -96,31 +53,38 @@ void Model::readRadiiAndAtomNumFromFile(std::string& filepath){
       elem_Z[substrings[1]] = std::stoi(substrings[0]);
     }
   }
-  return;
+  // Notify the user if no radius is defined
+  // the program can continue running because the user can manually define radii
+  if (radius_map.size() == 0) {
+    Ctrl::getInstance()->notifyUser("Invalid radii definition file!");
+    Ctrl::getInstance()->notifyUser("Please reload a valid file or define radii manually in the table.");
+  }
 }
 
 bool Model::readAtomsFromFile(std::string& filepath, bool include_hetatm){
 
-  std::vector<Atom> list_of_atoms;
   atom_amounts.clear();
+  raw_atom_coordinates.clear();
 
   if (fileExtension(filepath) == "xyz"){
-    readFileXYZ(list_of_atoms, filepath);
+    readFileXYZ(filepath);
   }
   else if (fileExtension(filepath) == "pdb"){
-    readFilePDB(list_of_atoms, filepath, include_hetatm);
+    readFilePDB(filepath, include_hetatm);
   }
   else { // The browser does not allow other file formats but a user could manually write the path to an invalid file
-    Ctrl::getInstance()->notifyUser("Invalid structure file format!");
+    Ctrl::getInstance()->notifyUser("!!!!!!\nInvalid structure file format!\n!!!!!!");
     return false;
   }
-
-  atoms = list_of_atoms;
-  storeAtomsInTree();
+  if (raw_atom_coordinates.size() == 0){ // If no atom is detected in the input file, the file is deemed invalid
+    Ctrl::getInstance()->notifyUser("!!!!!!\nInvalid structure file!\n!!!!!!");
+    return false;
+  }
+  Ctrl::getInstance()->notifyUser("Structure file loaded successfully.");

```

Comment:
I think this is self evident from the table. This clutters my output text box.

Reply:
I agree. I wanted to make sure that a user who might have inadvertently not seen the "invalid structure file!" error when it was displayed first would not think it is an error related to his currently loaded file.
When you repeat some actions with the program, it is difficult to differentiate what output line is related to what event.
For example, running 10 calculations with different parameters, I have to check the duration of the different calculations to navigate through the output text (as the duration is the only unique marker).

Reply:
I feel like the whole way we currently handle the output to the user is unrefined and just preliminary anyway. I mean there's tonnes of problems with the output. It's hard to read, new outputs are added to the bottom of the list which I have to manually scroll down to, There is no visual distinction between errors and messages, the output never resets.

I think the whole way we handle outputs need a complete overhaul. But I also think this is better handled towards the end of the project, when we have a better idea as to what the user will need to see. For now, I'd only make sure that we have the basic infrastructure for user communication and optimise it for our use and for debugging.